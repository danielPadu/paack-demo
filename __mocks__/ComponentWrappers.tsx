import {NavigationContext} from '@react-navigation/native';
import React from 'react';
import {Provider} from 'react-redux/src';
import {PersistGate} from 'redux-persist/integration/react';
import {persistor} from '../appFiles/appStore/store';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import {defaultReducer} from '../appFiles/appStore/reducers/default';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
export const mockedStore = mockStore(defaultReducer);

export const navContext = {
  isFocused: () => true,
  // addListener returns an unscubscribe function.
  addListener: jest.fn(() => jest.fn()),
  navigate: Function.prototype,
  setParams: Function.prototype,
  dispatch: Function.prototype,
  getParam: (param: {deliveryId?: string} & any, defaultValue: any) => {
    return defaultValue;
  },
};

export const ComponentWrappers = (props: {
  children: React.ReactNode;
  redux_store: any;
}) => {
  const {children, redux_store} = props;

  return (
    <>
      <Provider store={redux_store ?? mockedStore}>
        <PersistGate loading={null} persistor={persistor}>
          <NavigationContext.Provider value={navContext as any}>
            {children}
          </NavigationContext.Provider>
        </PersistGate>
      </Provider>
    </>
  );
};
