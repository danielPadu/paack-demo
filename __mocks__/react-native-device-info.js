export default {
  getModel: jest.fn(),
  getVersion: jest.fn(),
  getBuildNumber: jest.fn(),
  getBundleId: jest.fn(),
  hasNotch: jest.fn(),
};
