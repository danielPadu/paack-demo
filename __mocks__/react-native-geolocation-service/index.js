export default {
  addListener: jest.fn(),
  getCurrentPosition: () => ({
    coords: {
      accuracy: 0,
      altitude: 0,
      altitudeAccuracy: 0,
      heading: 0,
      latitude: 0,
      longitude: 0,
      speed: 0,
    },
  }),
  removeListeners: jest.fn(),
  requestAuthorization: jest.fn(),
  setConfiguration: jest.fn(),
  startObserving: jest.fn(),
  stopObserving: jest.fn(),
};
