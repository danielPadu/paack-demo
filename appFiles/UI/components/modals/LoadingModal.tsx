import React from 'react';
import {StyleSheet, View} from 'react-native';
import Modal from 'react-native-modal';
import * as Progress from 'react-native-progress';
import {useDispatch, useSelector} from 'react-redux';
//import {connect} from 'react-redux';
import {loadingModalClose} from '../../../appStore/actions/modalActions';
import {screenHeight, screenWidth} from '../../utils';
import { useIsFocused } from '@react-navigation/native';

const LoadingModal = () => {
  const {
    loadingModal: {isOpen},
  } = useSelector((state: any) => state.modalReducer);
  const dispatch = useDispatch();
  const {modalStyle, containerStyle} = styles;

  const modalClosing = () => {
    if (isOpen) {
      dispatch(loadingModalClose());
    }
  };

  return (
    <Modal
      animationIn="fadeIn"
      animationOut="fadeOut"
      animationInTiming={400}
      propagateSwipe={true}
      //transparent={true}
      isVisible={isOpen}
      backdropColor={'#75747452'}
      onBackdropPress={() => modalClosing()}
      style={modalStyle}
      deviceWidth={screenWidth}
      deviceHeight={screenHeight}>
      <View style={containerStyle}>
        <Progress.Circle
          size={90}
          indeterminate={true}
          borderWidth={10}
          borderColor={'gray'}
          //endAngle={0.75}
          thickness={5}
          strokeCap={'round'}
          fill={'transparent'}
        />
      </View>
    </Modal>
  );
};
const styles = StyleSheet.create({
  modalStyle: {
    flex: 1,
    margin: 0,
    marginBottom: 0,
  },
  containerStyle: {
    zIndex: 99999,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    backgroundColor: 'white',
    opacity: 0.8,
  },
});
/* const mapStateToProps = store => {
  const {theme} = store.appReducer;
  const {
    loadingModal: {isOpen, options},
  } = store.modalReducer;
  return {isOpen, options, theme};
};
const mapDispatchToProps = dispatch => {
  return {
    modalLoadingClose: () => dispatch(modalLoadingClose()),
  };
}; */
export default /* connect(mapStateToProps, mapDispatchToProps)( */ LoadingModal;
