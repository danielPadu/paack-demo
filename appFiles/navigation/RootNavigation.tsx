// RootNavigation.js
import * as React from 'react';
import {createNavigationContainerRef} from '@react-navigation/native';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
 
export const navigationRef = createNavigationContainerRef<RootStackParamList>();
export type RootStackParamList = {
  IntroScreen: {itemId?: string};
  DeliveriesScreen: {itemId?: string};
  DeliveryDetailsScreen: {deliveryId?: string; itemId?: string};
};
export type DeliveryScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'DeliveriesScreen'
>;
export type DeliveryDetailsScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'DeliveryDetailsScreen'
>;
export type IntroScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'IntroScreen'
>;



export const navigate = (name: keyof RootStackParamList, params?: unknown) => {
  if (navigationRef.isReady()) {
    // Perform navigation if the app has mounted
    console.log('navigationRef navigating to : ', name, );
    navigationRef.navigate(name as never, params as never);
  } else {
    // You can decide what to do if the app hasn't mounted
    // You can ignore this, or add these actions to a queue you can call later
    console.log('CAN NOT NAVIGATE ');
  }
};
 