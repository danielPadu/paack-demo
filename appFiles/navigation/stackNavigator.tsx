import React, {useEffect} from 'react';

import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {IntroScreen, DeliveriesScreen, DeliveryDetailsScreen} from '../screens';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {navigationRef, isReadyRef} from './RootNavigation';
//import MembershipActiveScreen from '../screens/components/MembershipActiveScreen';
import {useDispatch} from 'react-redux';
import {setAppActiveScreen} from '../appStore/actions/appActions';
export type RootStackParamList = {
  IntroScreen: undefined;
  DeliveriesScreen: undefined;
  DeliveryDetailsScreen: {deliveryId: string};
};
const Stack = createStackNavigator();
const defaultOptions = {
  animationEnabled: false,
  swipeEnabled: false,
};
const STACK_PROPS = (props: {
  initialRouteName: string;
  children: React.ReactNode | React.ReactNode[];
}) => (
  <Stack.Navigator
    screenOptions={{
      headerShown: true,
      title: '',
      headerStyle: {height: 30},
    }}
    initialRouteName={
      props.initialRouteName ? props.initialRouteName : undefined
    }>
    {props.children}
  </Stack.Navigator>
);

// Gets the current screen from navigation state
const getActiveRouteName = state => {
  const route = state?.routes[state?.index];
  if (route?.state) {
    // Dive into nested navigators
    return getActiveRouteName(route?.state);
  }
  return route;
};

const AppContainer = () => {
  const previousActiveRouteRef = React.useRef();
  const activeRouteRef = React.useRef();
  const dispatch = useDispatch();

  useEffect(() => {
    //  const { removeOnNotificationOpened, removeOnNotification, removeOnTokenRefresh } = onlySetTheListners(); // set listners for FCM and detach them when app closes
    const state = navigationRef.current?.getRootState();
    activeRouteRef.current = getActiveRouteName(state); // Save the initial route name
    return () => {
      console.log('AppContainer unmounted');
      if (isReadyRef?.current) {
        (isReadyRef.current as React.MutableRefObject<boolean>).current = false;
      }
    };
  }, []);

  const theme = {
    //like this
    colors: {
      // background: "transparent",
      opacity: 0.75,
    },
  };
  return (
    <SafeAreaProvider>
      <NavigationContainer
        // theme={theme}
        // initialState={initialState}
        ref={navigationRef}
        // removeClippedSubviews={false}
        onStateChange={state => {
          const previousActiveRoute = activeRouteRef.current;
          const currentRoute = getActiveRouteName(state);
          if (previousActiveRoute !== activeRouteRef) {
            // console.log('NavigationContainer previousActive: -> ', previousActiveRoute);
            console.log('NavigationContainer active: -> ', currentRoute);
            dispatch(setAppActiveScreen(currentRoute.name));
          }
          // Save the current route name for later comparision
          activeRouteRef.current = currentRoute;
          previousActiveRouteRef.current = previousActiveRoute;
        }}
        onReady={() => {
          if (isReadyRef?.current) {
            (isReadyRef?.current as React.MutableRefObject<boolean>).current =
              true;
          }
        }}>
        <STACK_PROPS initialRouteName="IntroScreen">
          <Stack.Screen
            name="IntroScreen"
            component={IntroScreen}
            options={() => ({
              backBehavior: 'order',
              ...defaultOptions,
            })}
            initialParams={{itemId: 'Intro'}}
          />
          <Stack.Screen
            name="DeliveriesScreen"
            component={DeliveriesScreen}
            options={() => ({
              backBehavior: 'order',
              ...defaultOptions,
            })}
            initialParams={{itemId: 'DeliveriesScreen'}}
          />
          <Stack.Screen
            name="DeliveryDetailsScreen"
            component={DeliveryDetailsScreen}
            options={() => ({
              backBehavior: 'order',
              ...defaultOptions,
            })}
            initialParams={{itemId: 'DeliveryDetailsScreen'}}
          />
        </STACK_PROPS>
      </NavigationContainer>
    </SafeAreaProvider>
  );
};

export default AppContainer;
