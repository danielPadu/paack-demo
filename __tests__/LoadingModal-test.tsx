import React from 'react';
import renderer, {
  act,
  ReactTestRenderer,
  ReactTestRendererJSON,
} from 'react-test-renderer';
import LoadingModal from '../appFiles//UI/components/modals/LoadingModal';
import {ComponentWrappers, mockedStore} from '../__mocks__/ComponentWrappers';
import {navigate} from '../appFiles/navigation/RootNavigation';
import {waitFor} from 'react-native-testing-library';
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest
    .fn()
    .mockReturnValueOnce({loadingModal: {isOpen: true}})
    .mockReturnValueOnce({loadingModal: {isOpen: true}}),
  useDispatch: () => jest.fn(),
}));
let store;
let component:
  | ReactTestRenderer
  | ReactTestRenderer[]
  | ReactTestRendererJSON
  | ReactTestRendererJSON[]
  | null;
describe('checking LoadingModal component', () => {
  test('LoadingModal renders correctly', async () => {
    await act(async () => {
      await beforeEach(() => {
        store = mockedStore;

        component = renderer
          .create(
            <ComponentWrappers redux_store={store}>
              <LoadingModal />
            </ComponentWrappers>,
          )
          .toJSON();
        expect(component).toMatchSnapshot();
        expect(navigate('Deliveries')).toBeFalsy();
      });
    });
  });
  test('LoadingModal inner functionallity tests', async () => {
    await act(async () => {
      await beforeEach(async () => {
        store = mockedStore;

        component = renderer
          .create(
            <ComponentWrappers redux_store={store}>
              <LoadingModal />
            </ComponentWrappers>,
          )
          .toJSON();
        const {findByProps} = component;
        const indicator = findByProps('size');
        expect(component.props.modalClosing()).toEqual(true);
        await waitFor(() => expect(indicator).toBeTruthy());
        await waitFor(() => expect(component).toHaveProperty('containerStyle'));
      });
      await afterEach(() => {
        // cleanup on exiting
        component?.remove?.();
        component = null;
      });
    });
  });
});
