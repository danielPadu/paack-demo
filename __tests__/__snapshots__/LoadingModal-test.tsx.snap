// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking LoadingModal component LoadingModal inner functionallity tests 1`] = `
<Modal
  animationType="none"
  deviceHeight={1334}
  deviceWidth={750}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={1334}
    deviceWidth={750}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 0,
        "marginBottom": 0,
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "bottom": 0,
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "opacity": 0.8,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 99999,
        }
      }
    >
      <View
        indeterminateAnimationDuration={1000}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={90}
          bbWidth={90}
          collapsable={false}
          focusable={false}
          height={90}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                ],
              },
              Object {
                "flex": 0,
                "height": 90,
                "width": 90,
              },
            ]
          }
          width={90}
        >
          <RNSVGGroup
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
          >
            <RNSVGPath
              d="M45 5
          A40 40 0 1 1 21.488589908301066 12.639320225002109"
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                ]
              }
              stroke={4286611584}
              strokeLinecap={1}
              strokeWidth={10}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</Modal>
`;
